testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -120,6 +120,13 @@ def test_last_slice(self):
         frames = [ts.frame for ts in trj_iter]
         assert_equal(frames, np.arange(self.universe.trajectory.n_frames))
 
+    def test_pickle_singleframe_reader(self):
+        reader = self.universe.trajectory
+        reader_p = pickle.loads(pickle.dumps(reader))
+        assert_equal(len(reader), len(reader_p))
+        assert_equal(reader.ts, reader_p.ts,
```

Comment:
How does equality testing for `ts` work? Did we implement `__eq__`?? Does it compare floats anywhere? I'm just surprised that it works....

Reply:
```python
    def __eq__(self, other):
        """Compare with another Timestep

        .. versionadded:: 0.11.0
        """
        if not isinstance(other, Timestep):
            return False

        if not self.frame == other.frame:
            return False

        if not self.n_atoms == other.n_atoms:
            return False

        if not self.has_positions == other.has_positions:
            return False
        if self.has_positions:
            if not (self.positions == other.positions).all():
                return False

        if not self.has_velocities == other.has_velocities:
            return False
        if self.has_velocities:
            if not (self.velocities == other.velocities).all():
                return False

        if not self.has_forces == other.has_forces:
            return False
        if self.has_forces:
            if not (self.forces == other.forces).all():
                return False

        return True
```
Not everything is tested, but at least a few basic ones.

Reply:
Thank you, that's all good. Nice.